### PR TITLE
Invoke-DbaQuery, introduce noexec

### LINKED ISSUE #8158: Invoke-DbaQuery: Support 'NOEXEC' mode
### Summarize Functionality

Add functionality to Invoke-DbaQuery to prefix the query supplied with [`NOEXEC`](https://docs.microsoft.com/en-us/sql/t-sql/statements/set-noexec-transact-sql?view=sql-server-ver15). This would allow the command to be used for parsing/compiling error checking, and help to round out the lack of `WhatIf` capabilities like most other commands have.

Full disclosure, I'm working on an Ansible collection for SQL Server management ([lowlydba.sqlserver](https://github.com/lowlydba/lowlydba.sqlserver)) that is Powershell Core compatible, heavily leaning on DBATools. This functionality would be great to support [`check_mode`](https://docs.ansible.com/ansible/2.3/playbooks_checkmode.html) for the query module.

### Is there a command that is similiar or close to what you are looking for?

Yes - Invoke-DbaQuery

### Technical Details

Prepending the query with `SET NOEXEC ON;` and then suffixing it with `SET NOEXEC OFF;` should work.

### PR DESCRIPTION
**Pull request by niphlod**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [x] New feature (non-breaking change, adds functionality, fixes #8158 )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/dataplat/appveyor-lab ?
 - [x] Unit test is included
 - [ ] Documentation
 - [ ] Build system

### Purpose
Support NOEXEC "out of the box"

### Approach
In order to keep the Verbose stream unlittered/untouched by Confirm/Whatif, a new switch needed to be added.
See the new tests to see what it can/should be useful for, albeit @lowlydba can chime in and request some changes.
So let's see if everyone agrees on the "interface" before merging  ^_^

### CODE REVIEW

**Review by potatoqualitee — APPROVED** [review]
nice change and code looks good to me! 👌🏼

File: internal/functions/Invoke-DbaAsync.ps1 [hunk 1]
```diff
@@ -177,7 +182,11 @@ function Invoke-DbaAsync {
         # Only execute non-empty statements
         $Pieces = $Pieces | Where-Object { $_.Trim().Length -gt 0 }
         foreach ($piece in $Pieces) {
-            $cmd = New-Object Microsoft.Data.SqlClient.SqlCommand($piece, $conn)
+            $runningStatement = $piece
+            if ($NoExec) {
+                $runningStatement = "SET NOEXEC ON; " + $piece + " SET NOEXEC OFF;"
```
Comment (andreasjordan) [inline internal/functions/Invoke-DbaAsync.ps1:187]: Does the second "SET" needs a ";" in front? Sorry, no time to test this at the moment.

Comment (niphlod) [reply internal/functions/Invoke-DbaAsync.ps1]: great catch, let's add it, should be incosequential for simple queries but better safe than sorry

**Comment by potatoqualitee** [comment]
Thanks, everyone 🥳

**Pull request merged**